src/build/service.js

```diff
@@ -143,7 +150,8 @@ class BuildService {
     // set up to skip other tasks if this tag already exists locally
     const dockerImages = await this.docker.listImages();
     // TODO: need docker image sha, if it exists (or set it later)
-    ctx.skip = dockerImages.some(image => image.RepoTags.indexOf(tag) !== -1);
+    ctx.skipBuild = dockerImages.some(image => image.RepoTags && image.RepoTags.indexOf(tag) !== -1);
+    ctx.skipPush = false; // TODO: Make this setable by a cli flag
```

Comment:
I like this -- in most cases, we will want to push, so it should be the default.

Reply:
In my head I was kinda thinking that it would default to false and then when you've built everything and are happy with things, then you run again with the `--push` option or whatever. At that point all of the build steps get skipped since the builds have already happened and then the push happens. I'm pretty happy either way though.

Reply:
That's what I was thinking until you reminded me yesterday that most users of minikube will still need to push.

By the way, we probably need to actually do a `docker pull` or list remote tags or something to check whether the revision is already built upstream, too.  Otherwise we'll be at risk of rebuilding and pushing over tags that are already in production.